Lib/unittest/test/test_case.py

```diff
@@ -1273,6 +1273,19 @@ def testAssertRaisesNoExceptionType(self):
         with self.assertRaises(TypeError):
             self.assertRaises((ValueError, object))
 
+    def testAssertRaisesRefcount(self):
```

Comment:
Given the new try/finally handling and the potential for unexpected behavior, I think the context manager mode needs to be checked as well.  I don't see a need for a separate test function, just add something like:

```
with self.assertRaises(ValueError):
    func()

self.assertEqual(refcount, sys.getrefcount(func))
```